### DISCUSSION TITLE
Validator showing duplicate messages

### DISCUSSION THREAD

**Discussion by popovicieduard** [adonisjs/core#1312] [Help] [answered]
When validating via the Standalone validator, I receive duplicate messages on my response

```
export default class RegisterValidator {
  public schema = schema.create({
    username: schema.string({}, [
      rules.required(),
      rules.alpha(),
      rules.unique({ table: 'users', column: 'username' }),
      rules.minLength(4),
      rules.maxLength(60),
    ])
  })
```

response:
```
{
    "username": [
        "username is required",
        "username is required"
    ]
}
```

If i comment the rules.required(). the username will still show that it is required, and, it is strange because only the required rule acts this way, does anybody know why ?

**Comment by ammezie** [comment, marked as answer]
By default `schema.string()` marks the field as required. So no need to specify that explicitly again.

**Reply by dajpes** [reply]
True, if you remove rules.required() you will see the error message once.

**Comment by popovicieduard** [comment]
Hmm, and what about updating ? How can I have optional rules when doing CRUD ?

**Reply by thetutlage** [reply]
It's covered in the docs. https://preview.adonisjs.com/guides/validator/usage#marking-fields-as-optional
